android/src/main/java/com/amplitude/android/plugins/AndroidNetworkConnectivityCheckerPlugin.kt

```diff
@@ -0,0 +1,26 @@
+package com.amplitude.android.plugins
+
+import com.amplitude.android.Configuration
+import com.amplitude.android.utilities.AndroidNetworkConnectivityChecker
+import com.amplitude.core.Amplitude
+import com.amplitude.core.events.BaseEvent
+import com.amplitude.core.platform.Plugin
+import kotlinx.coroutines.launch
+
+class AndroidNetworkConnectivityCheckerPlugin : Plugin {
+    override val type: Plugin.Type = Plugin.Type.Before
+    override lateinit var amplitude: Amplitude
+    private lateinit var networkConnectivityChecker: AndroidNetworkConnectivityChecker
+
+    override fun setup(amplitude: Amplitude) {
+        super.setup(amplitude)
+        networkConnectivityChecker = AndroidNetworkConnectivityChecker((amplitude.configuration as Configuration).context, amplitude.logger)
+    }
+
+    override fun execute(event: BaseEvent): BaseEvent? {
+        amplitude.amplitudeScope.launch(amplitude.storageIODispatcher) {
+            amplitude.configuration.isNetworkConnected = networkConnectivityChecker.isConnected()
```

Comment:
We don't need to do this each time. What if we combine the listener with this plugin, and each time there is a change we just update the variable and trigger the callback(Probably in that way we don't need the callback interface anymore).

Reply:
I think we cannot rely on network change callback considering the case where the APP is started when offline, there is no network change but is still not connected to network. 

IMO Network change callback is used to flush stored events when network is back while network checker is what we should call every time when track.

Reply:
Added `config.offline` and will check this every time.